lib/task_3/task3.dart

```diff
@@ -1,19 +1,19 @@
 // Реализуйте метод, который принимает строку слов, разделённых пробелами.
 // Задача — найти в данной строке числа и вернуть коллекцию num этих чисел.
 
-List<int> numbersInString(String text) {
+List<num> numbersInString(String text) {
```

Comment:
```suggestion
List<num> numbersInString(String text) {
  List<String> list = text.split(" ");
  List<num> num1 = [];
  for (var element in list) {
      num? number = num.tryParse(element);
      if (number == null) {
        continue;
      }
      num1.add(number);
  }
  return num1;
}
```

Предлагаю в цикле убрать проверку is num

Вместо этого будет проверка `number` на null.

Так мы избежим повторной операции парсинга.